### PR TITLE
Delete all combinations when removing product from shop

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Combinations wasn't being delete during shop selection for product i shop was unselected.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | #30751
| How to test?      |  refer to fixed ticket
| Possible impacts? |

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

**Comment by zuk3975** [comment]
⚠️ 
need to double check if this PR #30683 didn't already handle the same things.

File: src/Adapter/Product/Shop/CommandHandler/DeleteProductFromShopsHandler.php [hunk 1]
```diff
@@ -42,19 +44,37 @@ class DeleteProductFromShopsHandler implements DeleteProductFromShopsHandlerInte
...
+        $shopIds = $command->getShopIds();
+        $productId = $command->getProductId();
+        $this->productRepository->deleteFromShops($command->getProductId(), $shopIds);
+
+        if (!$this->productRepository->hasCombinations($command->getProductId())) {
+            return;
+        }
+
+        foreach ($shopIds as $shopId) {
+            $this->combinationMultiShopRepository->deleteByProductId($productId, ShopConstraint::shop($shopId->getValue()));
```
Comment (jolelievre) [inline src/Adapter/Product/Shop/CommandHandler/DeleteProductFromShopsHandler.php]: Even if both work, wouldn't it be more logical to start by deleting the combinations before deleting the product they're associated to?

File: src/Adapter/Product/Update/ProductShopUpdater.php [hunk 2]
```diff
@@ -226,4 +228,27 @@ private function copyImageAssociations(ProductId $productId, ShopId $sourceShopI
...
     }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $sourceShopId
+     * @param ShopId $targetShopId
+     *
+     * @return void
+     */
+    private function copyCombinations(ProductId $productId, ShopId $sourceShopId, ShopId $targetShopId, string $productType): void
```
Comment (jolelievre) [inline src/Adapter/Product/Update/ProductShopUpdater.php]: I think this should be handled now that this PR https://github.com/PrestaShop/PrestaShop/pull/30683 has been merged, I already fixed combinations copy from to shop to shop (which is why you have conflicts right now)
I didn't notice the delete wasn't handled though, good catch!

**Review by jolelievre — COMMENTED** [review]
Thanks @zuk3975 

some conflicts to fix, and behat tests would be nice also once your other blocking PR is merged

**Comment by zuk3975** [comment]
This PR should wait for https://github.com/PrestaShop/PrestaShop/pull/30585 to be merged and then behat test scnearios can be added and manual testing should be less confusing (because before that PR combinations list in multiple shops are not dislpayed well)

**Review by jolelievre — COMMENTED** [review]
Thanks @zuk3975 

Not sure how we can check this with behat though, unless we create a step
`And combination prosut1SBlack should not exist anymore for shop shop1` 

Then you can check accurately that the combination has been removed for the appropriate shop

Oh or maybe you plan on using the same steps as the on added in the combination IDS PR?

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Review by kpodemski — APPROVED** [review]

**Review by djoelleuch — APPROVED** [review]
Hello @zuk3975 !

It's QA :heavy_check_mark:  

[Untitled_ Jan 9, 2023 3_41 PM.webm](https://user-images.githubusercontent.com/115152460/211335349-12ab2f7b-6087-49e2-8de8-564bd7a0a5ea.webm)

Thank you for your PR  :rocket:  !

**Pull request merged**